src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java

```diff
@@ -253,22 +254,24 @@ public boolean logOut(boolean showPrompt) {
     return true;
   }
 
-  private Credential makeCredential(String accessToken, String refreshToken) {
-    Credential cred =
-        new GoogleCredential.Builder()
-            .setJsonFactory(jsonFactory)
-            .setTransport(transport)
-            .setClientSecrets(clientId, clientSecret)
-            .build();
-    cred.setAccessToken(accessToken);
-    cred.setRefreshToken(refreshToken);
-    return cred;
+  private Credential makeCredential(
+      String accessToken, String refreshToken, long expiresInSeconds) {
+    Credential credential = new GoogleCredential.Builder()
+        .setJsonFactory(jsonFactory)
+        .setTransport(transport)
+        .setClientSecrets(clientId, clientSecret)
+        .build();
+    credential.setAccessToken(accessToken)
+              .setRefreshToken(refreshToken)
+              .setExpiresInSeconds(expiresInSeconds);
+    return credential;
   }
 
   private void updateLoginState(GoogleTokenResponse tokenResponse)
       throws IOException, EmailAddressNotReturnedException {
     Credential credential = makeCredential(tokenResponse.getAccessToken(),
-                                           tokenResponse.getRefreshToken());
+                                           tokenResponse.getRefreshToken(),
+                                           tokenResponse.getExpiresInSeconds());
     String email = queryEmail(credential);
     Long expiryTime = System.currentTimeMillis() / 1000 + tokenResponse.getExpiresInSeconds();
     accountRoster.addAccount(new Account(email, credential, expiryTime));
```

Comment:
In fact, I'm going the opposite. `Account` is currently holding an inaccurate, duplicate `accessTokenExpiryTime` unnecessarily, and I think we can remove the field completely since `Credential` already records an accurate time.